pkg/filesystem/subvolume.go

```diff
@@ -52,6 +54,78 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	resource := "storageclusters"
+	storageclusters, err := clientsets.ListResourcesDynamically(ctx, "ocs.openshift.io", "v1", resource, clusterNamespace)
+	if err != nil || !strings.Contains(err.Error(), "could not find") {
+		logging.Fatal(err)
+	}
+	if len(storageclusters) == 0 {
+		logging.Info("resource %s was not found on the cluster", resource)
+	}
+
+	for _, storagecluster := range storageclusters {
+		spec, _, err := unstructured.NestedMap(storagecluster.Object, "spec", "externalStorage")
+		if err != nil {
+			fmt.Printf("Error accessing 'externalStorage': %v\n", err)
+			os.Exit(1)
+		}
+		if spec["enable"] != nil {
+			enable = spec["enable"].(bool)
+			return enable
+		}
+	}
+	return enable
+}
+
+// getExternalClusterDetails exports the CEPH_ARGS with required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	getsecret := fmt.Sprintf(`kubectl get secret rook-ceph-external-cluster-details -n %s -o=jsonpath='{.data.external_cluster_details}' |  base64 -d`, clusterNamespace)
+	var admin_id, admin_key, m string
+	external_cluster_details := exec.ExecuteBashCommand(getsecret)
+	if external_cluster_details == "" {
+		logging.Error(fmt.Errorf("external cluster details not found"))
+	}
+	external_cluster_details = strings.TrimPrefix(external_cluster_details, "[")
+	external_cluster_details = strings.TrimSuffix(external_cluster_details, "]")
+
+	// This is the sample data of secret decoded from external-cluster-details above.
+	//{"name": "rook-ceph-mon-endpoints", "kind": "ConfigMap",
+	//"data": {"data": "ceph-ci-vavuthuextdevyp1-7w8tg-u3w3fp-node1-installer=10.0.209.62:6789", "maxMonId": "0", "mapping": "{}"}},
+	// {"name": "rook-csi-cephfs-provisioner", "kind": "Secret", "data": {"adminID": "csi-cephfs-provisioner",
+	//"adminKey": "AQDSVvhlMnPvABAAHqbM73kgNKYtXi9hwfDaBg=="}}, {"name": "rook-csi-cephfs-node", "kind": "Secret", "data":
+	//{"adminID": "csi-cephfs-node", "adminKey": "AQDSVvhl58U7ABAAtRkNDbAD4f1PXB0fNYlqwQ=="}}
+	// To connect to the ceph-cluster we need the mon-ip(10.0.209.62:6789), the adminID(csi-cephfs-node) and
+	// the key(AQDSVvhl58U7ABAAtRkNDbAD4f1PXB0fNYlqwQ==)
+	split := strings.Split(external_cluster_details, "},")
```

Comment:
cant we use json.unmarhsal to a struct?

Reply:
struct will be bit complex, using map was a bit direct I guess. Cause the format pasted in comment is just the part of what is required. The whole output is bit more . Map will be easy to lookup rather than iterating through an array of struct. what do you say?

Reply:
i believe struct will be straight forward but you need to know what  struct to unmarshal to, string split and traversing through it is kind of error prone

Reply:
okay, I will try to have a look at the struct and see if that can be done.